sdks/java/core/src/main/java/org/apache/beam/sdk/fn/data/BeamFnDataGrpcMultiplexer.java

```diff
@@ -64,7 +75,8 @@ public BeamFnDataGrpcMultiplexer(
           baseOutboundObserverFactory) {
     this.apiServiceDescriptor = apiServiceDescriptor;
     this.receivers = new ConcurrentHashMap<>();
-    this.erroredInstructionIds = new ConcurrentHashMap<>();
+    this.poisonedInstructionIds =
+        CacheBuilder.newBuilder().expireAfterWrite(Duration.ofMinutes(20)).build();
```

Comment:
Should 20 minutes be pulled out to a top-level constant?

Reply:
Done